Gems/ROS2/Code/Source/Camera/ROS2CameraSensorEditorComponent.cpp

```diff
@@ -0,0 +1,188 @@
+                        "Camera's vertical (y axis) field of view in degrees.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &ROS2CameraSensorEditorComponent::m_width, "Image width", "Image width")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &ROS2CameraSensorEditorComponent::m_height, "Image height", "Image height")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &ROS2CameraSensorEditorComponent::m_colorCamera, "Color Camera", "Color Camera")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &ROS2CameraSensorEditorComponent::m_depthCamera, "Depth Camera", "Depth Camera");
+            }
+        }
+    }
+
+    void ROS2CameraSensorEditorComponent::Activate()
+    {
+        AzFramework::EntityDebugDisplayEventBus::Handler::BusConnect(this->GetEntityId());
+        AzToolsFramework::Components::EditorComponentBase::Activate();
+    }
+
+    void ROS2CameraSensorEditorComponent::Deactivate()
+    {
+        AzFramework::EntityDebugDisplayEventBus::Handler::BusDisconnect();
+        AzToolsFramework::Components::EditorComponentBase::Deactivate();
+    }
+
+    void ROS2CameraSensorEditorComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC("ROS2Frame"));
+    }
+
+    void ROS2CameraSensorEditorComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("ROS2SensorCamera"));
+    }
+
+    void ROS2CameraSensorEditorComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("ROS2SensorCamera"));
+    }
+
+    void ROS2CameraSensorEditorComponent::BuildGameEntity(AZ::Entity* gameEntity)
+    {
+        gameEntity->CreateComponent<ROS2::ROS2CameraSensorComponent>(
+            m_sensorConfiguration, m_VerticalFieldOfViewDeg, m_width, m_height, m_colorCamera, m_depthCamera);
+    }
+
+    void ROS2CameraSensorEditorComponent::DisplayEntityViewport(
+        const AzFramework::ViewportInfo& viewportInfo, AzFramework::DebugDisplayRequests& debugDisplay)
+    {
+        if (!m_sensorConfiguration.m_visualise)
+        {
+            return;
+        }
+        AZ_UNUSED(viewportInfo);
+        const AZ::u32 stateBefore = debugDisplay.GetState();
+        AZ::Transform transform = GetEntity()->GetTransform()->GetWorldTM();
+
+        // dimension of drawing
+        const float arrowRise = 1.1f;
+        const float arrowSize = 0.5f;
+        const float frustumScale = 0.1f;
+
+        transform.SetUniformScale(frustumScale);
+        debugDisplay.DepthTestOff();
+        const float ver = 0.5f * AZStd::tan((AZ::DegToRad(m_VerticalFieldOfViewDeg * 0.5f)));
+        const float hor = m_height * ver / m_width;
+
+        // frustum drawing
+        const AZ::Vector3 p1(-ver, -hor, 1.f);
```

Comment:
suggestion: I believe there might actually be a util already to do frustum drawing in O3DE (see `DisplayFrustum` in `AzFramework/Visibility/VisibilityDebug` - might be useful here potentially

Reply:
Yes, DisplayFrustum is very similar, but our code has an image-processing convention and uses the camera matrix as a parameter.  I think it will be complicated to use DisplayFrustum here.

Reply:
Okay no problem, just thought I'd mention it in case it was useful, thanks for letting me know 👍